### PR TITLE
Handle LinalgExt ops in dispatch region pretty names

### PR DESCRIPTION
**Pull request by silvasean**

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp [hunk 1]
```diff
@@ -31,18 +31,55 @@ namespace IREE {
...
 
-  int64_t cost = 1;
-  for (auto loopRange : op.getStaticLoopRanges()) {
-    cost *= loopRange;
+static TensorType getMainTensorForLinalgExtOp(Operation *op) {
+  TensorType main;
+  auto operandTypes = llvm::to_vector(op->getOperandTypes());
+  auto resultTypes = llvm::to_vector(op->getResultTypes());
+  for (Type t : llvm::concat<Type>(operandTypes, resultTypes)) {
+    if (auto tensorType = t.dyn_cast<TensorType>()) {
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: We can write it in an early-continue form, so we can save one level of nesting. E.g.,

```
auto tensorType = t.dyn_cast ...
if (!tensorType) continue;
// ...
```

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp [hunk 2]
```diff
@@ -31,18 +31,55 @@ namespace IREE {
...
+      if (!main) {
+        main = tensorType;
+      } else if (costOfDomain(tensorType.getShape()) >
+                 costOfDomain(main.getShape())) {
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: maybe cache the `costOfDomain(main.getShape())`, so we can have less computation?

**Review by hanhanW — APPROVED** [review]

**Comment by hanhanW** [comment]
Thanks, I was thinking for doing this for a while. :)

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp (hunk 2 above)
Comment (silvasean) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: I don't think it's significant enough to uglify the code.

Comment (hanhanW) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: That's true.. I think the LinalgExt ops are not even fusing with others in current pipeline, except set/unset_encodings.

**Pull request merged**